library/Core/src/Juvix/Core/IR/Evaluator.hs

```diff
@@ -231,6 +265,7 @@ toLambda (IR.GFunction (IR.Function {funUsage = π, funType = ty, funClauses}))
     toLambda' π (IR.quote ty) pats rhs
 toLambda _ = Nothing
 
+-- | Translate a `RawGlobal'` function definition into an elimination term.
```

Comment:
Basically, `toLambda[R]` doesn't assume that the global environment has the same extension as the term being inlined into. the ultimate reason for these `toOnlyExts{T,E}` functions is to be able to convert to another `ext`, which requires some conditions:

- the input ext can't have any extension constructors (the `NoExtensions` constraint), since it would not know how to translate them over
- the output ext can't have any annotations (the `OnlyExts` wrapper), since they can't necessarily be conjured out of thin air

there are other ways this could have been solved, of course, for example by simply requiring the globals to have the same ext as the current term, which in hindsight probably would have been better.

anyway, the new descriptions you wrote sound good to me.